web/book/src/transforms/sort.md

```diff
@@ -42,39 +42,75 @@ sort [s"substr({first_name}, 2, 5)"]
 
 ## Ordering guarantees
 
-In PRQL, a relation is an _array of tuples_ and not a set or a bag. The
-difference is that array has inherent ordering. This makes the order of the
-relation persist trough sub-queries and intermediate table definitions.
+Ordering is persistent through a pipeline in PRQL. For example:
 
-In SQL, on the other hand, relations do not have an order and can be ordered
-only in the context of query result, `TAKE` clause or window functions. This has
-a unexpected effect where a previously ordered relation would get reshuffled
-because of following JOIN or windowing operations.
+```prql
+from employees
+sort tenure
+join locations [==employee_id]
+```
+
+Here, PRQL pushes the `sort` down the pipeline, compiling the `ORDER BY` to the
+_end_ of the query. Consequently, most relation transforms retain the row order.
 
-For example, this query:
+The explicit semantics are:
+
+- `sort` introduces a new order,
+- `group` resets the order,
+- `join` maintains the order of the left relation,
+- database tables don't have a defined order.
```

Comment:
Well, they do. Postgres will return the same order of rows, no matter how many times you execute a query (if you are not modifying the table in any way).

It's just that order is unknown.

(I'm a bit pedantic here, but people reading "semantics" are usually like that)

Reply:
Pedantic is good here! I switched it

Reply:
Pedantic is good here! I switched it